broken_link_checker/checker.py

```diff
@@ -83,15 +76,30 @@ def __init__(self, host: str, delay: int = 1, deep_scan=False):
             re.IGNORECASE
         )
 
-    def check(self, url: str) -> urllib3.response.HTTPResponse:
+    def is_same_host(self, url):
+        """
+        Verify if the url belongs the host.
+
+        :url the url to verify
+        """
+        host = requests.utils.urlparse(self.host)
+        url = requests.utils.urlparse(url)
+
+        if not url.scheme:
+            return True
+        elif url.scheme == host.scheme\
+         and url.netloc == host.netloc\
+           and url.port == host.port:
+            return True
+        else:
+            return False
```

Comment:
```suggestion
        return True if str(self.host) in str(url) else False
```

Is not this enought 🤔  ??

Reply:
The idea is to know if a URL belongs a host.
- relative URL belongs the host
- absolute URL belongs only if the scheme and netloc are same with the host.